sdk/java-sdk/src/main/java/kalix/javasdk/valueentity/ValueEntity.java

```diff
@@ -56,6 +58,17 @@ public void _internalSetCommandContext(Optional<CommandContext> context) {
     commandContext = context;
   }
 
+  /** INTERNAL API */
+  public void _internalSetCurrentState(S state) {
+    currentState = Optional.ofNullable(state);
+  }
+
+  protected final S currentState() {
+    return currentState.orElseThrow(
+        () ->
+            new IllegalStateException("Current state can only available when handling a command."));
+  }
```

Comment:
We may need to introduce this new API, because otherwise it will be quite alien to have entity methods with Rest annotations, but with params that are not participating on the REST API.

For example: 

```java
 @PostMapping("/user/{id}/create")
    public ValueEntity.Effect<Done> createUser(User user, @RequestBody CreateUser createUser) {
      return effects().reply(Done.instance);
    }
```

In that example, `User` can only be passed by Kalix and doesn't not participate on the REST API.

Reply:
Can we have intermediate component baseclasses in the Spring SDK rather than extend the core Java SDK with features only needed for the Spring SDK?

Reply:
I consider this option as well, but the drawback is that users will have access to the same classes of the Java SDK. 
We will have to duplicate them (not an issue) and then we will have: `kalix.javasdk.valueentity.ValueEntity` and  `kalix.springsdk.valueentity.ValueEntity` to import. 

But may not be a major issue:
* we can emit a nice message showing that the wrong type was used. Also, if users pick the Java SDK one, they won't have access to the state, so they should realise it quite soon. If they try have a method receiving the State, it won't work neither. 
* I think we will need a `@KalixComponent` annotation anyway. If possible, we should restrict it's usage to some Spring SDK interface. If not possible, we can do it with annotations processors at compile time. I'm not talking about generated code, but to fail compilation if we see that `@KalixComponent` is being used in a class that is one of the allowed ones.